crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -203,6 +204,37 @@ impl<'flox> Publish<'flox, Empty> {
 }
 
 impl<'flox> Publish<'flox, NixAnalysis> {
+    /// Sign the binary
+    ///
+    /// Requires a valid signing key
+    pub async fn sign_binary(
+        self,
+        key_file: impl AsRef<Path>,
+    ) -> Result<Publish<'flox, NixAnalysis>, PublishError> {
+        let nix = self.flox.nix(Default::default());
+        let installable = Installable {
+            flakeref: self.publish_flake_ref.clone().into_inner(),
+            attr_path: self.attr_path.clone(),
+        };
+
+        let sign_command = StoreSign {
+            store_sign: StoreSignArgs {
+                key_file: key_file.as_ref().into(),
+                recursive: Some(true.into()),
+            },
+            installables: [installable].into(),
```

Comment:
I'm not quite sure what was the reason not to sign by installable in the bash implementation...
However, this impl is not entirely correct either, as we'd want to sign all outputs assumably but can't represent output selectors `^*` in runix.
I'd call this impl blocked on flox/runix#30 on top which we can implement outputs or sidestep that issue for a while longer and use store paths directly

Reply:
flox/runix#41